developer-docs-site/docs/standards/fungible-asset.md

```diff
@@ -0,0 +1,403 @@
+    /// be displayed to a user as `5.05` (`505 / 10 ** 2`).
+    decimals: u8,
+}
+```
+
+### Fungible Asset and Fungible Store
+
+FA allows typing to be decoupled from metadata by allocating an object reference that points at the metadata. Hence a set of units of FA is represented as an amount and a reference to the metadata, as shown:
+
+```rust
+struct FungibleAsset {
+    metadata: Object<Metadata>,
+    amount: u64,
+}
+```
+
+In contrast to Objects and addresses, a Coin uses a generic, or the `CoinType`, to support distinct typing within the Coin framework. For example, `Coin<A>` and `Coin<B>` are two distinct coins, if `A != B`.
+
+The fungible assets is a struct representing the type and the amount of units held. As the struct does not have either key or store abilities, it can only be passed from one function to another but must be consumed by the end of a transaction. Specifically, it must be deposited back into a fungible store at the end of the transaction:
+
+```rust
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct FungibleStore has key {
+    /// The address of the base metadata object.
+    metadata: Object<Metadata>,
+    /// The balance of the fungible metadata.
+    balance: u64,
+    /// Fungible Assets transferring is a common operation, this allows for freezing/unfreezing accounts.
+    frozen: bool,
+}
+```
+
+The only extra field added here is `frozen`. if it is `true`, this object is frozen, i.e. deposit and withdraw are both disabled without using `TransferRef` in the next section.
+
+### References
+
+_Reference_ (ref) is the means to implement permission control across different standards in Aptos. In different contexts, it may be called _capabilities_. The FA standard has three distinct refs for minting, transferring, and burning FA: `MintRef`, `TransferRef`, and `BurnRef`. Each ref contains a reference to the FA metadata:
+
+```rust
+struct MintRef has drop, store {
+    metadata: Object<Metadata>
+}
+
+struct TransferRef has drop, store {
+    metadata: Object<Metadata>
+}
+
+struct BurnRef has drop, store {
+    metadata: Object<Metadata>
+}
+```
+
+Ref owners can do the following operations depending on the refs they own:
+
+- `MintRef` offers the capability to mint new FA units.
+- `TransferRef` offers the capability to mutate the value of `freeze` in any `FungbibleStore` of the same metadata or transfer FA by ignoring `freeze`.
+- `BurnRef` offers the capability to burn or delete FA units.
+
+The three refs collectively act as the building blocks of various permission control system as they have `store` so can be passed around and stored anywhere. Please refer to the source file for `mint()`, `mint_to()`, `burn()`, `burn_from()`, `withdraw_with_ref()`, `deposit_with_ref()`, and `transfer_with_ref()`: These functions are used to mint, burn, withdraw, deposit, and transfer FA using the MintRef, BurnRef, and TransferRef.
```

Comment:
May be we can give a quick example here taken from managed_fungible_asset where these refs are stored to be used by the creator later